src/shared/scaling/scaleArray.js

```diff
@@ -1,20 +1,45 @@
-export default function scaleArray(array, scalingParameters) {
+export default function scaleArray(
+  imageFrame,
+  preScaleOptions,
+  array,
+  scalingParameters
+) {
   const arrayLength = array.length;
-  const { rescaleSlope, rescaleIntercept, suvbw } = scalingParameters;
+  const rescaleSlope = scalingParameters?.rescaleSlope || 1;
+  const rescaleIntercept = scalingParameters?.rescaleIntercept || 0;
+  const suvbw = scalingParameters?.suvbw || 1;
+
+  let isNegative = false;
 
   if (scalingParameters.modality === 'PT') {
     if (typeof suvbw !== 'number') {
       return;
     }
 
     for (let i = 0; i < arrayLength; i++) {
-      array[i] = suvbw * (array[i] * rescaleSlope + rescaleIntercept);
+      const value = suvbw * (array[i] * rescaleSlope + rescaleIntercept);
+
+      array[i] = value;
+      if (value < 0) {
```

Comment:
wouldn't it be faster if we don't check on value < 0 if the isNegative is `true` too ?